lib/samples/query_related_features/query_related_features.dart

```diff
@@ -0,0 +1,280 @@
+
+    // Feature table for related preserves layer
+    _alaskaNationalParksPreservesTable = ServiceFeatureTable.withUri(
+      Uri.parse(
+        'https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/ArcGIS/rest/services/AlaskaNationalParksPreservesSpecies_List/FeatureServer/0',
+      ),
+    );
+    // Feature table for related species layer
+    _alaskaNationalParksSpeciesTable = ServiceFeatureTable.withUri(
+      Uri.parse(
+        'https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/ArcGIS/rest/services/AlaskaNationalParksPreservesSpecies_List/FeatureServer/2',
+      ),
+    );
+    // Add these to the tables on the map.
+    map.tables.addAll(
+      [_alaskaNationalParksSpeciesTable, _alaskaNationalParksPreservesTable],
+    );
+
+    // assign map to the map view
+    _mapViewController.arcGISMap = map
+      ..initialViewpoint = Viewpoint.fromCenter(
+        ArcGISPoint(
+          x: -16507762.575543,
+          y: 9058828.127243,
+          spatialReference: SpatialReference.webMercator,
+        ),
+        scale: 36764077,
+      );
+
+    // set selection color
+    _mapViewController.selectionProperties =
+        SelectionProperties(color: Colors.yellow);
+
+    // Set the ready state variable to true to enable the sample UI.
+    setState(() => _ready = true);
+  }
+
+  void onTap(Offset offset) async {
+    // Clear the selection on the feature layer.
+    _alaskaNationalParksLayer.clearSelection();
+
+    // Do an identify on the feature layer and select a feature.
+    final identifyLayerResult = await _mapViewController.identifyLayer(
+      _alaskaNationalParksLayer,
+      screenPoint: offset,
+      tolerance: 12.0,
+      maximumResults: 1,
+    );
+
+    // If there are features identified, show the bottom sheet to display the
+    // attachment information for the selected feature.
+    final features =
+        identifyLayerResult.geoElements.whereType<Feature>().toList();
+    if (features.isNotEmpty) {
+      _alaskaNationalParksLayer.selectFeatures(features: features);
+      final selectedFeature = features.first as ArcGISFeature;
+      setState(() {
+        _layerDataVisible = true;
+        _loadingFeatures = true;
+      });
+      // Query for related features.
+      queryRelatedFeatures(selectedFeature);
+    } else {
+      setState(() {
+        _layerDataVisible = false;
+        _loadingFeatures = false;
+      });
+    }
+  }
+
+  // Query for related features given the origin feature.
+  void queryRelatedFeatures(ArcGISFeature selectedPark) async {
+    // Query for related features.
+    final selectedParkTable = selectedPark.featureTable as ServiceFeatureTable;
+    final relatedFeatureQueryResult =
+        await selectedParkTable.queryRelatedFeatures(feature: selectedPark);
+
+    // Get the related species and preserves features.
+    final lists = <List<String>>[];
+    for (final RelatedFeatureQueryResult result in relatedFeatureQueryResult) {
+      final list = <String>[];
```

Comment:
Could we rename the `list` and `lists` variables to be more descriptive? Where we end up with `lists.add(list);` it's quite hard to have a grasp on what they actually contain.

Reply:
Yes for sure!